src/blueprints/tv2/timeline-object-factories/tv2-caspar-cg-timeline-object-factory.ts

```diff
@@ -308,6 +317,52 @@ export class Tv2CasparCgTimelineObjectFactory implements Tv2GraphicsElementTimel
     }
   }
 
+  public createAudioBedTimelineObject(audioBedName: string, blueprintConfiguration: Tv2BlueprintConfiguration): CasparCgMediaTimelineObject {
+    const audioBedSettings: AudioBedSettings = blueprintConfiguration.studio.audioBedSettings
+    const audioBedConfiguration: AudioBedConfiguration | undefined = blueprintConfiguration.showStyle.audioBedConfigurations.find(audioBedConfiguration => audioBedConfiguration.name === audioBedName)
+    if (!audioBedConfiguration) {
+      throw new NotFoundException(`Unable to find the audio bed configuration for ${audioBedName}.`)
+    }
+    return {
+      id: 'casparCg_audio_bed',
+      enable: {
+        start: 0
+      },
+      layer: Tv2CasparCgLayer.AUDIO,
+      priority: 1,
+      content: {
+        deviceType: DeviceType.CASPAR_CG,
+        type: CasparCgType.MEDIA,
+        file: this.assetPathHelper.joinAssetToFolder(audioBedConfiguration.filename, audioBedSettings.mediaDirectory),
+        channelLayout: AUDIO_CHANNEL_LAYOUT,
+        loop: true,
+        noStarttime: true,
+        mixer: {
+          volume: audioBedSettings.volume / 100,
+        },
+        transitions: {
+          inTransition: {
+            type: CasparCgTransitionType.MIX,
+            easing: CasparCgTransitionEase.LINEAR,
+            direction: CasparCgTransitionDirection.LEFT,
+            duration: this.convertFramesToMilliseconds(audioBedConfiguration.fadeInDurationInFrames),
+          },
+          outTransition: {
+            type: CasparCgTransitionType.MIX,
+            easing: CasparCgTransitionEase.LINEAR,
+            direction: CasparCgTransitionDirection.LEFT,
+            duration: this.convertFramesToMilliseconds(audioBedConfiguration.fadeOutDurationInFrames),
+          },
+        }
+      },
+      classes: ['lyd_on_air'],
+    }
+  }
+
+  private convertFramesToMilliseconds(frames: number): number {
+    return 1000 * frames / FRAME_RATE
+  }
```

Comment:
We should come up with a way so we don't have to have this helper function in all our TimelineObjectFactories

Reply:
I have created a separate PR for it, to make the changes easier to review: https://github.com/tv2/sofie-server/pull/197